Documentation/ApiOverview/Database/RestrictionBuilder/Index.rst

```diff
@@ -223,3 +228,51 @@ deliver and use an own set of restrictions for own query statements if needed.
    It can be very helpful to debug the final statements created by the `RestrictionBuilder` using
    :php:`debug($queryBuilder->getSQL())` right before the final call to :php:`$queryBuilder->execute()`. Just
    take care these calls **do not** :ref:`end up in production <database-query-builder-get-sql>` code.
+
+
+A very complex example is that you might want to apply one or more restriction/s to only one table. Let's say,
+that you have content in table `tt_content` that could have another record as parent record assigned in field
+`parent`. Now you would like to get all records except the child records where the parent record is hidden.
+
+   // use TYPO3\CMS\Core\Utility\GeneralUtility;
+   // use TYPO3\CMS\Core\Database\ConnectionPool;
+   // use TYPO3\CMS\Core\Database\Query\Restriction\DeletedRestriction;
+   // use TYPO3\CMS\Core\Database\Query\Restriction\EndTimeRestriction;
+   // use TYPO3\CMS\Core\Database\Query\Restriction\StartTimeRestriction;
+   // use TYPO3\CMS\Core\Database\Query\Restriction\DefaultRestrictionContainer;
+
+   $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)->getQueryBuilderForTable('tt_content');
+   
+   // Remove all restrictions first
+   $queryBuilder
+      ->getRestrictions()
+      ->removeAll()
+   
+   // build a custom restriction container
+   $myRestrictionContainer = new DefaultRestrictionContainer();
+   
+   // remove all restricitons that should not been used
+   // with this, only the HiddenRestriction stays in the restriction container
+   $myRestrictionContainer
+      ->removeByType(DeletedRestriction::class)
+      ->removeByType(EndTimeRestriction::class)
+      ->removeByType(StartTimeRestriction::class);
+   
+   // apply the restrictions in "$myRestrictionContainer" only to one table
+   $doctrineQueryBuilderExpressions = $myRestrictionContainer->buildExpression(['c2' => 'tt_content'], $queryBuilder->expr());
+
+   // add those restrictions to the orWhere query part
+   $queryBuilder
+      ->select('c1.*')
+      ->from('tt_content', 'c1')
+      ->leftJoin('c1', 'tt_content', 'c2', 'c1.parent = c2.uid')
+      ->orWhere($queryBuilder->expr()->isNull('c2.uid'), $doctrineQueryBuilderExpressions);
+
+With this you will end up with the following query:
```

Comment:
Again, make sure the next block is formatted as code block. But here we cannot use PHP so we should use:

> With this you will end up with the following query:
>
> .. code-block: sql